src/components/LetterSelection/index.jsx

```diff
@@ -1,9 +1,130 @@
-import React from 'react';
+import React, { useReducer, useEffect } from 'react';
+import {
+  makeSelectableLetters,
+  makeBlankLetters,
+} from '../../utilities/letterSelectionUtil';
+
+const types = {
+  prevLetter: 'prevLetter',
+  addLetter: 'addLetter',
+  removeLetter: 'removeLetter',
+  initialize: 'initialize',
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case types.prevLetter: // decrements counter in letterState
+      return {
+        ...state,
+        letterCounter: state.letterCounter === 0 ? 0 : state.letterCounter - 1,
+      };
+
+    case types.addLetter: // checks if clicked letter's index is already in clickedLetterIndexes of the letterState and if letterCounter is lower than the word's length, then replaces empty strings in clickedLetters array with letter, "pushes" index of letter to clickedLetterIndexes, and increments the counter
+      if (
+        !state.clickedLetterIndexes.includes(action.payload.index) &&
+        state.letterCounter < state.clickedLetters.length
+      ) {
+        return {
+          ...state,
+          letterCounter: state.letterCounter + 1,
+          clickedLetters: state.clickedLetters.map((letter, index) => {
+            if (index === state.letterCounter) {
+              letter = action.payload.letter;
+            }
+            return letter;
+          }),
+          clickedLetterIndexes: [
+            ...state.clickedLetterIndexes,
+            action.payload.index,
+          ],
+        };
+      } else {
+        return state;
+      }
+
+    case types.removeLetter:
+      const clickedLetterIndexesPopped = state.clickedLetterIndexes.slice(
+        0,
+        -1
+      );
+      return {
+        ...state,
+        letterCounter: state.letterCounter === 0 ? 0 : state.letterCounter - 1,
+        clickedLetters: state.clickedLetters.map((letter, index) => {
+          if (index + 1 === state.letterCounter) {
+            letter = '';
+          }
+          return letter;
+        }),
+        clickedLetterIndexes: clickedLetterIndexesPopped,
+      };
+
+    case types.initialize: // initalize the state with resetting letterCounter, new selectable letters, and new blank strings
+      return {
+        ...state,
+        letterCounter: 0,
+        selectableLetters: makeSelectableLetters(action.payload),
+        clickedLetters: makeBlankLetters(action.payload),
+        clickedLetterIndexes: [],
+      };
+
+    default:
+      return state;
+  }
+};
+
+const initialLetterState = {
+  letterCounter: 0,
```

Comment:
Could you elaborate? Currently the `letterCounter` is used as a pointer for where the selection of the letter would be. The length of the `clickedLetterIndexes` would just be a static number.

Reply:
![Screen Shot 2021-07-31 at 3 22 04 PM](https://user-images.githubusercontent.com/6326891/127751636-c38a9eef-42e2-4c66-9178-f7550352c077.png)
It seems as those two numbers will always be the same no? Just may eliminate duplicate data and margin for error, though again it could also be harder to edit and think about.

Reply:
Ah I see. My mistake. You're right, the length of the clickedLetterIndexes will correspond to the letterCounter. Though I think I'd like to keep the counter because it's easier to understand.